tests/MassTransit.S3.Storage.Tests/Storage_Specs.cs

```diff
@@ -0,0 +1,75 @@
+namespace MassTransit.S3.Storage.Tests
+{
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Amazon.S3;
+    using Microsoft.Extensions.DependencyInjection;
+    using NUnit.Framework;
+    using S3Storage.MessageData;
+    using TestFramework;
+    using Testing;
+
+
+    public class Storage_Specs
+    {
+        IServiceProvider _provider;
+
+        [SetUp]
+        public void Setup()
+        {
+            Environment.SetEnvironmentVariable("AWS_REGION", "us-east-1");
+            Environment.SetEnvironmentVariable("AWS_ACCESS_KEY_ID", "admin");
+            Environment.SetEnvironmentVariable("AWS_SECRET_ACCESS_KEY", "admin");
+            MessageDataDefaults.TimeToLive = TimeSpan.FromDays(2);
+            MessageDataDefaults.ExtraTimeToLive = TimeSpan.FromMinutes(5);
+            var dataRep = new S3MessageDataRepository(new AmazonS3Config()
+            {
+                ForcePathStyle = true,
+                ServiceURL = "http://localhost:4566",
+            }, "test");
+            _provider = new ServiceCollection()
+                .AddMassTransitTestHarness(x =>
+                {
+                    x.SetKebabCaseEndpointNameFormatter();
+
+                    x.UsingInMemory((context, cfg) =>
+                    {
+                        cfg.ConnectBusObserver(dataRep);
+                        cfg.UseMessageData(dataRep);
```

Comment:
I'd like to see one of the new configuration methods that other repositories have so that the repository is configured/built with the bus instead of outside. This is a new repository, so there isn't any reason to have a legacy builder with new(), etc.

Reply:
Not entirely sure but i think you meant? if so new changes are made
```c#
 cfg.UseMessageData(d => d.AmazonS3Storage("test", new AmazonS3Config()
        {
            ForcePathStyle = true,
            ServiceURL = "http://localhost:4566",
        }));
```

Reply:
Yep, that's what I meant :)